### ISSUE TITLE
Cannot render plots for Hugo/blogdown .Rmd

### ISSUE THREAD

**Issue by palfalvi** [jbkunst/highcharter#283]
Hi,

I wanted to use highcharter with blogdown+Hugo pages. When I made the Rmd file, in RStudio everything worked well, but in the rendered webpage there is just a blank space where the chart should go. I tried to put it in a htmlwidgets::tagList() but the same. I am using simple highchart() with two columnrange hc_add_series() element. Do you have any solution for this? 

Thank you in advance.

**Comment by jbkunst** [comment]
Hi @palfalvi,

I don't know how blogdown work. So:

- Have you tried with other htmlwidgets?
- In the page when you see a blank space, what do you see in the devtools (F12 button)?

**Comment by palfalvi** [comment]
Other htmlwidgets looks fine (I am continously working with DiagrammeR, DT, networkD3, plotly).

On html the blank space is:

```
<div id="htmlwidget-ea368e20fb13e05444cd" style="width:100%;height:500px;" class="highchart html-widget html-widget-static-bound"></div>

<script type="application/json" data-for="htmlwidget-ea368e20fb13e05444cd">...</script>
```
"..." has the data values for rendering the plot. 

On debug there are this messages for highchart.js:

```
Line 62:63 $("#" + el.id).highcharts is not a function. (In '$("#" + el.id).highcharts(x.hc_opts)', '$("#" + el.id).highcharts' is undefined)
Line 98:42 $("#" +el.id).highcharts is not a function. (In '$("#" +el.id).highcharts()', '$("#" +el.id).highcharts' is undefined)
```

**Comment by jbkunst** [comment]
Usually that message is when highcharts.js is not [loaded](http://stackoverflow.com/questions/32011248/highcharts-uncaught-typeerror-highcharts-is-not-a-function) or maybe there's an issue with jquery. To be sure, have you see if highcharts the source (CRTL+U)  is correctly loaded? 

Do you have an open repo with your blogdown site?

**Comment by palfalvi** [comment]
Looks like the sources are loaded, just that 2 lines have problem. Thank for the link, however how can I load JQuery before Highchart.js from R? 

I have an [open repo](https://github.com/palfalvi/palfalvi.github.io), but I did not uploaded any post with highcharts so far since it cannot show anything on local.

**Comment by palfalvi** [comment]
Btw on [your site under shiny](http://jkunst.com/highcharter/shiny.html) also I cannot see any plot. Can it caused by similar issues?

**Comment by jbkunst** [comment]
Can you try with a simple theme? I want to be sure if  there some conflicts with the scripts loaded at the end

**Comment by palfalvi** [comment]
I tried with a simple chart using ```favorite_pies``` dataset with several build and custom themes, including ```hc_theme_null()```, but still the same.

**Comment by jbkunst** [comment]
Sorry, I wasn't clear enough. I was talking about the hugo theme ;)

**Comment by palfalvi** [comment]
You was right. I built a new site locally and just inserted a highcharter plot in the example .Rmd file and it was visible and fine. 

However I don't know is it worth to rearrange the whole site and experience with other themes until it is working. Probably it should be done in the beginning, but most likely it should work with all themes. Do you have any idea what's wrong in the theme then?

**Comment by jbkunst** [comment]
Nice!

My first guess is think in a conflict with the theme js and highcharts. You can try the new widgetframe by @bhaskarvk, I think **worth** a try, tell me if you try this approach :D.

(You're not alone https://github.com/rstudio/blogdown/issues/20)

**Comment by palfalvi** [comment]
Ok I tried several things. First of all, simple ```widgetFrame()``` was not working. 

However I realized ```plotly``` is also not working, but not all the time (?). Some plots are shown but some are not. ```wordcloud2``` makes also just an empty box. Better to search in the theme or blogdown, then in the widgets. Thanks for helping!

**Comment by jbkunst** [comment]
I agree, in that case it seems to be a blogdown issue. Meanwhile I'll close the issue, if you need more highcharter support we can reopen or make a new issue here ;)!

**Comment by AlexSemenovIRL** [comment]
I was following this ticket too, thank you for the update. 
Cheers,
Pork Chop

**Comment by paulditterline** [comment]
So glad I found this thread. Using ghostwriter theme for my blogdown blog and no charts when I push, although they show up on locally. Seems I need to switch themes!

**Comment by jbkunst** [comment]
I will look into it because that's not the idea! So it's seems correct to reopen the issue.

**Comment by bhaskarvk** [comment]
Rightnow widgetframe works with every other html output format except blogdown. I have an open issue for this https://github.com/rstudio/blogdown/issues/38. This is not something I can easily fix in widgetframe, but something to be handled at blogdown end.

**Comment by jbkunst** [comment]
Thanks @bhaskarvk for the update!

**Comment by bhaskarvk** [comment]
widgetframe now works with blogdown too.

**Comment by jbkunst** [comment]
Hey @bhaskarvk! Thanks for let us know!  We miss you!

**Comment by houstonhaynes** [comment]
FWIW I sort of have the widgetframe package working with highcharter, but there is a kludge because the URL construct is wrong inside of framewidget.

**Comment by feddelegrand7** [comment]
Just use the amazing widgetframe package.

**Comment by houstonhaynes** [comment]
https://github.com/bhaskarvk/widgetframe/issues/18#issue-345021897
